### PR TITLE
8250961: Move Universe::update_heap_info_at_gc to CollectedHeap

### PR DESCRIPTION
**Pull request by adityamandaleeka**
This is a small refactoring that moves `update_heap_info_at_gc()` and related members to `CollectedHeap`. There should be no behavioral change.

jdk/hotspot tier1 passed

JBS: https://bugs.openjdk.java.net/browse/JDK-8250961

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Issue
 * [JDK-8250961](https://bugs.openjdk.java.net/browse/JDK-8250961): Move Universe::update_heap_info_at_gc to CollectedHeap

### Reviewers
 * [Stefan Karlsson](https://openjdk.java.net/census#stefank) (@stefank - **Reviewer**) ⚠️ Review applies to 64d1c399b5940ccf2be932a994a96ab160b8528e
 * [Kim Barrett](https://openjdk.java.net/census#kbarrett) (@kimbarrett - **Reviewer**)

### Download
`$ git fetch https://git.openjdk.java.net/jdk pull/27/head:pull/27`
`$ git checkout pull/27`

### CODE REVIEW

File: src/hotspot/share/gc/shared/collectedHeap.hpp
Comment (stefank) [inline src/hotspot/share/gc/shared/collectedHeap.hpp:443]: These should be initialized in the constructor.

**Review by stefank — COMMENTED** [review]
Now that the variables and functions have moved to CollectedHeap, I don't think we need the _heap_ infix anymore.

Maybe rename variables to be named:
_capacity_at_last_gc
_used_at_last_gc

Functions:
free_at_last_gc()
used_at_last_gc()
update_capacity_and_used_at_gc()

File: src/hotspot/share/gc/shared/collectedHeap.hpp
Comment (stefank) [reply src/hotspot/share/gc/shared/collectedHeap.hpp:443]: Move these up to the other variables.

File: src/hotspot/share/gc/shared/collectedHeap.hpp
Comment (stefank) [inline src/hotspot/share/gc/shared/collectedHeap.hpp:524]: Maybe move these to after unused(). Alternatively, move it near soft_ref_policy(), since this is currently only used by soft ref handling.

**Comment by adityamandaleeka** [comment]
@stefank Thanks for the suggestions. I've updated the PR.

File: src/hotspot/share/gc/shared/collectedHeap.hpp
Comment (stefank) [inline src/hotspot/share/gc/shared/collectedHeap.hpp:248]: There are some unnecessary spaces between () and {. Could you also make these functions const?

**Review by stefank — APPROVED** [review]
Thanks. Just two small nits below. You still need a second reviewer before this can be integrated.

**Comment by kimbarrett** [comment]
Looks good except for Stefan's comments about free/used_at_last_gc().

**Comment by stefank** [comment]
@kimbarrett you need to use the Review button to make this officially reviewed.
@tschatzl FYI, will not get listed as a Reviewer if you don't review on the GitHub pages.

**Review by tschatzl — CHANGES_REQUESTED** [review]
@author: Same as @stefank mentioned, please do the requested cleanup.

@stefank : I can't talk about @kimbarrett , but I was simply waiting on the author to do the change and then approve (on github).

Now officially reviewed with a "request changes" result which will hold up integration (?).

**Comment by kimbarrett** [comment]
/reviewer add kimbarrett

Reviewed on the assumption the requested trivial modification will be made.

**Review by kimbarrett — APPROVED** [review]
I'm assuming the small changes to free/used_at_last_gc will be made.

**Comment by adityamandaleeka** [comment]
Thanks for the reviews @stefank @tschatzl @kimbarrett . I pushed an update addressing the last comments.

**Review by kimbarrett — APPROVED** [review]

**Comment by stefank** [comment]
Thanks. Could you merge in master, make sure that this still builds, and run the jtreg tests in test/hotspot/jtreg/gc? After that you can add a comment with /integrate, and I'll sponsor this patch for you.

**Comment by stefank** [comment]
@adityamandaleeka Your branch is still 34 commits behind openjdk:master. It doesn't look like you merged the master branch into your 8250961. Or did you only do that locally?

**Comment by adityamandaleeka** [comment]
@stefank Yes, I did the merge/build/test locally and didn't push anything since I saw the comment from the bot above saying that the change would be auto-rebased when merging since there are no conflicts with my branch.

**Comment by stefank** [comment]
This caused a failure building without precompiled headers. @kimbarrett is handling that with JDK-8252995.

**Pull request closed without merging**
